src/resolver.js

```diff
@@ -50,4 +55,40 @@ export default {
 
     yield createElements(routeMatches, fetchedComponents, fetchedData);
   },
+
+  /**
+   * Generate route data according to their getters, respecting the order of
+   * promises per the `defer` flag on routes.
+   */
+  getData(match, routeMatches) {
+    return accumulateRouteValues(
+      routeMatches,
+      match.routeIndices,
+      ({ ancestorRouteData, prevParentPromise }, routeMatch) => {
+        // For a deferred route, the parent promise is the previous promise.
+        // Otherwise, it's the previous parent promise.
+        const parentPromise = routeMatch.route.defer
+          ? Promise.all(ancestorRouteData)
```

Comment:
This is a little different... I think it makes more sense to wait on _all_ parent promises, instead of just the most immediate one.

Reply:
I think my intention there was that some child routes would not have to wait for the promises they don't care about.

For example, one could have a root route, which fetches some general data, such as recent news maybe, which are required on each page. This promise does not hold any other route from fetching its' own data.

Then, you could have another route, let's say `/admin`, which requires the user to be authenticated, and hence it fetches the auth state in its own `getData`. And finally, the route which is included into `/admin`, e.g. `/admin/news`, needs to wait till authentication state will be loaded, but **does not** need to wait till recent news are fetched.

But if I understood correctly, in your example the chain would be 
Fetch news => Fetch auth state => Fetch admin news
But in my desired behaviour it would be something like
Fetch news && (Fetch auth state => Fetch admin news)

Hope I could express this clearly :))

Reply:
I know what you mean, but I think that makes things a little unclear. What I mean is, suppose the route config looks like:

```js
<Route {...} getData={getData1}>
  <Route {...} getData={getData2}>
    <Route {...} getData={getData3} defer />
  </Route>
</Route>
```

Suppose `getData1` and `getData2` both return promises.

Now suppose we change `getData2` to no longer return a promise. With the earlier implementation, `getData3` would switch to waiting on `getData1` instead of `getData2`. This seems a bit more confusing then if it switched from waiting on both `getData1` and `getData2` to only waiting on `getData1`.

In the case above, I'd consider handling things like "recent news" with a named child route, perhaps, such that the hierarchy only includes immediate parents anyway... something like:

```js
<Route {...} getData={getAuthData}>
  {{
    recentNews: <Route {...} getData={getRecentNews} />,
    private: <Route {...} getData={getPrivateData} defer />,
  }}
</Route>
```

How does that sound?

Reply:
Ah, I see now. Then I haven't got any more objections. I think we can merge this